internal/controller/controllers/clusterdeployments_controller_test.go

```diff
@@ -244,6 +244,23 @@ var _ = Describe("cluster reconcile", func() {
 
 				validateCreation(cluster)
 			})
+
+			It("pull secret in string data", func() {
```

Comment:
not sure if this test is needed, you are testing k8s functionality and not the controller.